oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/AsyncIndexUpdate.java

```diff
@@ -426,6 +428,31 @@ private void checkIfStopped() throws CommitFailedException {
 
     @Override
     public synchronized void run() {
+        boolean isIndexWithLanePresent = false;
+        NodeState oakIndexNode = store.getRoot().getChildNode("oak:index");
+        if (!oakIndexNode.exists()) {
+            log.info("lane: {} not present for indexes under /oak:index", name);
+            return;
+        }
+        for (ChildNodeEntry childNodeEntry : oakIndexNode.getChildNodeEntries()) {
+            if (childNodeEntry.getNodeState().getProperty("async") != null) {
+                Iterator<String> a = childNodeEntry.getNodeState().getProperty("async").getValue(Type.STRINGS).iterator();
+                while (a.hasNext()) {
+                    if (a.next().equals(name)) {
```

Comment:
further simplification
```suggestion
            PropertyState async = childNodeEntry.getNodeState().getProperty("async");
            if (async != null) {
                for (String s : async.getValue(Type.STRINGS)) {
                    if (s.equals(name)) {
```

Reply:
done.